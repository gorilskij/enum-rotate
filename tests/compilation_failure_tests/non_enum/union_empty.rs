use derive_enum_rotate::EnumRotate;

#[derive(EnumRotate)]
union Union {}
