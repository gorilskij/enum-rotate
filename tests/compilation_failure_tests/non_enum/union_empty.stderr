error: Union Union is not an enum, EnumRotate can only be derived for enums
 --> tests/compilation_failure_tests/non_enum/union_empty.rs:4:1
  |
4 | union Union {}
  | ^^^^^^^^^^^^^^

error: unions cannot have zero fields
 --> tests/compilation_failure_tests/non_enum/union_empty.rs:4:1
  |
4 | union Union {}
  | ^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/compilation_failure_tests/non_enum/union_empty.rs:4:15
  |
4 | union Union {}
  |               ^ consider adding a `main` function to `$DIR/tests/compilation_failure_tests/non_enum/union_empty.rs`
